File: repos_estudiantes_TARQUI_segunda_entrega/RUBIN_DE_CELIS_VALDEZ_ANDRES/HERENCIA/EJERCICIO-1/Vehiculo.java
public class Vehiculo {
    private String marca;
    private String modelo;
    private int año;
    private double precio_base;

    public Vehiculo(String marca, String modelo, int año, double precio_base) {
        this.marca = marca;
        this.modelo = modelo;
        this.año = año;
        this.precio_base = precio_base;
    }
    
    public void mostrar_info() {
        System.out.println("Marca: " + marca);
        System.out.println("Modelo: " + modelo);
        System.out.println("Anio: " + año);
        System.out.println("Precio base: $" + precio_base);
    }

    public String getMarca() {
        return marca;
    }

    public void setMarca(String marca) {
        this.marca = marca;
    }

    public String getModelo() {
        return modelo;
    }

    public void setModelo(String modelo) {
        this.modelo = modelo;
    }

    public int getAño() {
        return año;
    }

    public void setAño(int año) {
        this.año = año;
    }

    public double getPrecioBase() {
        return precio_base;
    }

    public void setPrecioBase(double precio_base) {
        this.precio_base = precio_base;
    }
}

File: repos_estudiantes_ALVAREZ_segunda_entrega/Chino_Blanco_Noemi/1Herencia/Ejercicio1/Vehiculo.py
class Vehiculo:
    def __init__(self, marca, modelo, anio, precio_base):
        self.marca = marca
        self.modelo = modelo
        self.anio = anio
        self.precio_base = precio_base

    def mostrar_info(self):
        return f"Marca: {self.marca}, Modelo: {self.modelo}, Año: {self.anio}, Precio base: ${self.precio_base}"

class Coche(Vehiculo):
    def __init__(self, marca, modelo, anio, precio_base, num_puertas, tipo_combustible):
        super().__init__(marca, modelo, anio, precio_base)
        self.num_puertas = num_puertas
        self.tipo_combustible = tipo_combustible

    def mostrar_info(self):
        base_info = super().mostrar_info()
        return f"{base_info}, Puertas: {self.num_puertas}, Combustible: {self.tipo_combustible}"

class Moto(Vehiculo):
    def __init__(self, marca, modelo, anio, precio_base, cilindrada, tipo_motor):
        super().__init__(marca, modelo, anio, precio_base)
        self.cilindrada = cilindrada
        self.tipo_motor = tipo_motor

    def mostrar_info(self):
        base_info = super().mostrar_info()
        return f"{base_info}, Cilindrada: {self.cilindrada}, Motor: {self.tipo_motor}"

# SIN GETTERS NI SETTERS


vehiculos = [
    Coche("Toyota", "Corolla", 2025, 25000, 4, "Gasolina"),
    Coche("Ford", "Explorer", 2024, 35000, 5, "Diesel"),
    Moto("Yamaha", "R3", 2025, 7000, "321cc", "4 tiempos"),
    Moto("nisan", "CBR600RR", 2023, 12000, "599cc", "4 tiempos")
]

print(" Información de Vehículos:")
for v in vehiculos:
    print(v.mostrar_info())


print("\n Coches con más de 4 puertas:")
for v in vehiculos:
    if isinstance(v, Coche) and v.num_puertas > 4:
        print(v.mostrar_info())


print("\n Vehículos del año 2025:")
for v in vehiculos:
    if v.anio == 2025:
        print(v.mostrar_info())
